### PR TITLE
Wallet warning on Athlete page

### PR DESCRIPTION
**Pull request by MichaelFrank0317**
# Description
For better UX User should get a warning popup if they don't connect to the wallet and try to Buy / Sell / Mint

Fixes Jira Ticket # 630
https://athletex.atlassian.net/browse/AX-630

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Is this a UI Change? If so please include screenshot of before and after states below
image

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have assigned 2 reviewers to check my work
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/pages/athlete/components/buttons.dart [hunk 1]
```diff
@@ -8,12 +8,32 @@ import 'package:ax_dapp/pages/scout/dialogs/athlete_page_dialogs.dart';
...
+    title: Text(
+      'Wallet Warning',
+      style: TextStyle(color: Colors.amber[400], fontSize: 18),
+    ),
+    description: Text(
+      'You need to connect Wallet to use this feature.',
+      style: TextStyle(color: Colors.amber[400]),
+    ),
+  ).show(context);
+}
```
Comment (narcodico) [inline lib/pages/athlete/components/buttons.dart:36]: Please move this for time being into utils and make it an extension method on `BuildContext`.

**Review by narcodico — CHANGES_REQUESTED** [review]

File: lib/pages/athlete/components/buttons.dart (hunk 1 above)
Comment (MichaelFrank0317) [reply lib/pages/athlete/components/buttons.dart]: I made a extension method on utils

File: lib/pages/athlete/components/buttons.dart [hunk 2]
```diff
@@ -69,7 +51,7 @@ Container buyButton(
             ),
           );
         } else {
-          showWarningToast(context);
+          WalletWarning(context).show();
```
Comment (narcodico) [inline lib/pages/athlete/components/buttons.dart:54]: calling this should be `context.showWalletWarningToast()`

File: lib/util/wallet_warning_toast.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        'Wallet Warning',
+        style: TextStyle(color: Colors.amber[400], fontSize: 18),
+      ),
+      description: Text(
+        'You need to connect Wallet to use this feature.',
+        style: TextStyle(color: Colors.amber[400]),
+      ),
+    ).show(this);
+  }
+}
```
Comment (narcodico) [inline lib/util/wallet_warning_toast.dart:22]: this is not quite what I was saying.

```
extension ToastX on BuildContext {
  void showWalletWarningToast() => MotionToast(...).show(this);
}
```

**Review by narcodico — CHANGES_REQUESTED** [review]

**Review by narcodico — APPROVED** [review]

**Pull request merged**